web_src/js/features/captcha.js

```diff
@@ -0,0 +1,42 @@
+import {isDarkTheme} from '../utils.js';
+
+export function initCaptcha() {
+  const captchaEl = document.querySelector('#captcha');
+  if (!captchaEl) return;
+
+  const siteKey = captchaEl.getAttribute('data-sitekey');
+  const isDark = isDarkTheme();
+
+  const params = {
+    sitekey: siteKey,
+    theme: isDark ? 'dark' : 'light'
+  };
+
+  switch (captchaEl.getAttribute('captcha-type')) {
+    case 'g-recaptcha': {
+      // eslint-disable-next-line no-undef
+      if (grecaptcha) {
```

Comment:
I think you can avoid the eslint disables by using `window.` prefix.

Reply:
I'm currently in the countryside for the Chinese New Year holiday, and I don't have the right equipment for me to write the code, I'll fix it tomorrow or the day after tomorrow.

Reply:
No hurry. Another method may be to replace the script tags with `await import(url)` from JS. That the return value of the captcha scripts are in local scope and no globals are needed.

Reply:
I have modified the code. The captcha's global variables will be fetched from `window`.

Reply:
Thanks, looks good. If the providers support integration without relying on globals (by using url `import`), that would be the ideal integration, but this is good enough.